astroid/rebuilder.py

```diff
@@ -1354,7 +1360,8 @@ def visit_attribute(
             newnode = nodes.AssignAttr(node.attr, node.lineno, node.col_offset, parent)
             # Prohibit a local save if we are in an ExceptHandler.
             if not isinstance(parent, nodes.ExceptHandler):
-                self._delayed_assattr.append(newnode)
+                # mypy doesn't recognize that newnode has to be AssignAttr
+                self._delayed_assattr.append(newnode)  # type: ignore [arg-type]
```

Comment:
I would suggest to add that it's because mypy doesn't understand `ParamSpec` yet.